src/python/zquantum/core/cost_function.py

```diff
@@ -1,6 +1,8 @@
 from .interfaces.cost_function import CostFunction
 from .interfaces.backend import QuantumBackend
+from .interfaces.ansatz import Ansatz
 from .circuit import build_ansatz_circuit
+from .utils import create_symbols_map
 from typing import Callable, Optional, Dict
 import numpy as np
 import copy
```

Comment:
I don't see copy being used anywhere

Reply:
Adressed in [73b8143](https://github.com/zapatacomputing/z-quantum-core/pull/42/commits/73b8143e1c5870115ce7f99c9d94978711159e66) (I think?)